board/ipaccess/ipa73xx/ipa73xx.c

```diff
@@ -129,6 +129,37 @@ DECLARE_GLOBAL_DATA_PTR;
 
 /* Functions --------------------------------------------------------------- */
 
+
+/* Storage of ecc_check_result in SRAM:
+ *   0x2001ffac - ecc_check_result reversed copy 2
+ *   0x2001ffa8 - ecc_check_result copy 2
+ *   0x2001ffa4 - ecc_check_result reversed copy 1
+ *   0x2001ffa0 - ecc_check_result copy 1
+ */
+#define SRAM_PK_ECC_RESULT_ADDRESS 0x2001ffa0
+
+static void load_pk_ecc_result(void)
+{
+    ulong r0, r1, r2, r3;
+    volatile ulong* pk_ecc_result_addr = (volatile ulong*)(SRAM_PK_ECC_RESULT_ADDRESS);
+    
+    r3 = ~(pk_ecc_result_addr[3]);
+    r2 = pk_ecc_result_addr[2];
+    r1 = ~(pk_ecc_result_addr[1]);
+    r0 = pk_ecc_result_addr[0];
+    
+    if ( (r0 == r1) && (r0 == r2) && (r0 == r3) )
+    {
+        char result_str[11];
+        snprintf(result_str, sizeof(result_str), "%d", (int)r0);
```

Comment:
OK
10000: No key present 
10001: Invalid key (uncorrectable errors)
10002: Invalid result (values in memory don't match)
But I still don't understand why, since we are reading from hard-coded memory locations that are specifically 32 bits. Do I need to go right down to the SPL to change this, or stick with the current memory values and just interpret them in u-boot?